File: N Squared Sorts/C++/bubbleSort.cpp
void bubbleSort(int* arr, int n) {
  bool sorted;
  for (unsigned int i = 0; i < n - 1; i++) {
    sorted = true;
    for (unsigned int j = 0; j < n - 1 - i; j++) {
      if (arr[j] > arr[j + 1]) {
        std::swap(arr[j], arr[j + 1]);
        sorted = false;
      }
    }
    if (sorted) break;
  }
}

File: N Squared Sorts/python/bubble_sort.py
def bubbleSort(array):
    end = len(array) - 1
    Sorted = False
    while not Sorted:
        Sorted = True
        for i in range(end):
            if array[i] > array[i + 1]:
                array[i], array[i + 1] = array[i + 1], array[i]
                Sorted = False
        end -= 1
    return array
